src/states/joining_node.rs

```diff
@@ -361,6 +360,22 @@ impl JoiningNode {
             let _ = self.crust_service.disconnect(*peer_id);
         }
     }
+
+    fn compute_new_id(&mut self, name_range: &(XorName, XorName)) -> Result<(), RoutingError> {
+        let now = Instant::now();
+        loop {
+            if now.elapsed() > Duration::from_secs(COMPUTING_IDENTITY_TIMEOUT_SECS) {
```

Comment:
Lets hope the client/joining node is not busy, or it might not manage many iterations. Often this type of code allows a maximum number of tries instead of a timeout.